ntbs-service/Services/AuthorizationService.cs

```diff
@@ -29,7 +35,39 @@ public async Task<bool> IsUserAuthorizedToManageAlert(ClaimsPrincipal user, Aler
             return userTbServiceCodes.Contains(alert.TbServiceCode);
         }
 
-        public async Task<bool> CanEdit(ClaimsPrincipal user, Notification notification)
+        public IEnumerable<NotificationBannerModel> SetFullAccessOnNotificationBanners(
+            IEnumerable<NotificationBannerModel> notificationBanners,
+            ClaimsPrincipal user)
+        {
+            return notificationBanners.Select(async n =>
+            {
+                if(n.NotificationStatus != NotificationStatus.Legacy)
+                {
+                    var fullAccess = await CanEditBannerModel(user, n);
+                    n.FullAccess = fullAccess;
+                }
+                return n;
+            })
+            .Select(n => n.Result);
+        }
+
+        public async Task<bool> CanEditBannerModel(ClaimsPrincipal user, NotificationBannerModel notificationBannerModel)
+        {
+            var tbServiceCode = notificationBannerModel.TbServiceCode;
+            var tbServicePhecCode = notificationBannerModel.TbServicePHECCode;
+            var locationPhecCode = notificationBannerModel.LocationPHECCode;
+            return await GetUserPermissionFromLocationAndService(user, tbServiceCode, locationPhecCode, tbServicePhecCode);

```

Comment:
Hmm, quite a mouthful..
If we call it something like `AuthorizeUserAccess`, will that capture the intent well enough?